pageserver/src/walreceiver.rs

```diff
@@ -182,7 +182,7 @@ async fn wal_receiver_main_thread_loop_step<'a>(
             info!("Processing timeline update: {update:?}");
             match update {
                 // Timeline got detached, stop all related tasks and remove public timeline data.
-                LocalTimelineUpdate::Detach(id) => {
+                LocalTimelineUpdate::Detach(id, join_sender) => {
```

Comment:
It's a bit hard to understand what is this sender for before coming here, so a named enum field might be a bit more helpful.

If I understand it right, it's used once only to ensure that the drop happened, so we can presumably use a `oneshot` flavor from the stdlib?

Reply:
AFAIK there is no separate oneshot channel in std. You probably confused it with the tokio one. The default synchronous mpsc from std is oneshot by default and is transparently upgraded: https://doc.rust-lang.org/src/std/sync/mpsc/mod.rs.html#713-716

Reply:
But I agree, naming can be improved

Reply:
Changed it to be more verbose